djangocms_text_ckeditor/static/djangocms_text_ckeditor/js/cms.ckeditor.js

```diff
@@ -51,51 +52,223 @@
+        },
+
+        initInlineEditors: function () {
+            if (CMS._plugins === undefined) {
+                // no plugins -> no inline editors
+                return;
+            }
+
+            CMS.CKEditor.observer = CMS.CKEditor.observer || new IntersectionObserver(function (entries, opts) {
+                entries.forEach(function (entry ){
+                    if (entry.isIntersecting) {
+                        var target = $(entry.target);
+                        var plugin_id = target.data('plugin_id');
+                        var url = target.data('edit_url');
+                        CMS.CKEditor.startInlineEditor(plugin_id, url);
+                    }
+                });
+            }, {
+                root: null,
+                threshold: 0.05
+            });
 
-                // add additional styling
-                CKEDITOR.on('instanceReady', $.proxy(CMS.CKEditor, 'setup'));
+            CMS._plugins.forEach(function (plugin) {
+                if (plugin[1].plugin_type === 'TextPlugin') {
+                    var url = plugin[1].urls.edit_plugin;
+                    var id = plugin[1].plugin_id;
+                    var elements = $('.cms-plugin.cms-plugin-' + id);
+                    var wrapper;
+
+                    if (elements.length > 0) {
+                        if (elements.length === 1 && elements.prop('tagName') === 'DIV') {  // already wrapped?
+                            wrapper = elements.addClass('cms-ckeditor-inline-wrapper');
+                        } else {  // no, wrap now!
+                            wrapper = elements
+                                .wrapAll('<div class="cms-ckeditor-inline-wrapper wrapped"></div>')
+                                .parent();
+                            elements
+                                .removeClass('cms-plugin')
+                                .removeClass('cms-plugin-' + id);
+                            wrapper.addClass('cms-plugin').addClass('cms-plugin-' + id);
+                        }
+                        wrapper.data('edit_url', url);
+                        wrapper.data('plugin_id', id);
+                        wrapper.data('placeholder_id', plugin[1].placeholder_id);
+                        wrapper.on('dblclick', function (event) {
+                            event.stopPropagation();
+                        });
+                        wrapper.on('pointerover', function (event) {
+                            event.stopPropagation();
+                        });
+                        CMS.CKEditor.observer.observe(wrapper[0]);
+                    }
+                }
+            });
+        },
+
+        startInlineEditor: function (plugin_id, url) {
+            var options = {},
+                settings = JSON.parse(document.getElementById('ck-cfg-' + plugin_id).textContent),
+                wrapper = $('.cms-plugin.cms-plugin-' + plugin_id);
+
+            if (wrapper.data('ckeditor-initialized')) {
```

Comment:
it's unlikely to have much of a performance effect, but maybe easier to check if it's initialized in CMS.CKEditor.editors?